portal/templates/profile_macros.html

```diff
@@ -366,7 +368,87 @@ <h4 class="text-success registration-label">registered</h4>
     {% endif %}
 {%- endmacro %}
+{% macro profileRegistrationEmail(person) -%}
+    {%- if person -%}
+    <div id="sendRegistrationEmailForm">
+        <input type="hidden" name="_userId" id="_userId" value="{{person.id}}" />
+        <div class="registration-status-container">
+            <span class="text-muted">Registration status: </span>
+            {% if person and person.has_role("write_only") %}<h4 class="text-warning registration-label">not yet registered</h4>
+            {%- else -%}
+            <h4 class="text-success registration-label">registered</h4>
+            {%- endif -%}
+        </div>
+        <button id="btnProfileSendEmail" class="btn btn-tnth-primary disabled">{{ _("Send registration email") }}</button>
+        <div id="profileEmailMessage" class="text-info"></div>
+        <div id="profileEmailErrorMessage" class="text-danger"></div>
+    </div>
+    <script>$(function () {
+        function getAccessUrl() {
+            var url = '';
+            $.ajax ({
+                type: 'GET',
+                url: '/api/user/' + $("#_userId").val() + '/access_url',
+                contentType: "application/json; charset=utf-8",
+                dataType: 'json',
+                async: false
+            }).done(function(data) {
+                url = data['access_url'];
+                $("#profileEmailErrorMessage").text("");
+                console.log(url);
+            }).fail(function() {
+                $("#profileEmailErrorMessage").text("failed request to get email invite url");
+            });
+            return url;
+        };
+        if (!hasValue($("#email").val())) $("#profileEmailSelect").attr("disabled", true);
 
+        $('#btnProfileSendEmail').on('click', function(event) {
+            event.preventDefault();
+            var email = $('#email').val();
+            var subject = '';
+            var body = '';
+            var return_url = '';
+            var clinicName = (function() {
+                var orgs = $("#userOrgs input[name='organization']:checked");
+                var parentName = "";
+                if (orgs.length > 0 ) {
+                    orgs.each(function() {
+                        if (!parentName) {
+                            parentName = $(this).attr("data-parent-name");
+                            if (!hasValue(parentName)) parentName = $(this).closest(".org-container[data-parent-id]").attr("data-parent-name");
+                        };
+                    });
+                };
+                var cn = parentName ? parentName: "your clinic";
+                return cn;
+            })();
+            return_url = getAccessUrl();
+            if (hasValue(return_url)) {
+                body = "<p>Hello,</p>";
+                body += "<p>This email was sent to you from " + clinicName + ". This is an invitation to complete your registration.</p>";
```

Comment:
hmm... clinicName is generated dynamically from client side here, not sure if I can apply gettext in this context